seed/views/portfoliomanager.py

```diff
@@ -296,6 +296,29 @@ def get_template_by_name(templates, template_name):
         _log.debug("Desired report name found, template info: " + json.dumps(matched_template, indent=2))
         return matched_template
 
+    def parse_template_response(self, response_text):
+        """
+        This method is for the updated ESPM where the response is escaped JSON string in a JSON response.
+
+        :param response_text: str, repsonse to parse
+        :return: dict
+        """
+        try:
+            # the data are now in the string of the data key of the returned dictionary with an excessive amount of
+            # escaped doublequotes.
+            # e.g., response = {"data": "{"customReportsData":"..."}"}
+            decoded = json.loads(response_text)  # .encode('utf-8').decode('unicode_escape')
```

Comment:
in the future, we should centralize some of this functionality or move to puppeteer.